### PR TITLE
Support the generic ml ppl command.

### PR DESCRIPTION
**Pull request by jngz-es**
Signed-off-by: Jing Zhang <jngz@amazon.com>

### Description
Add a new method run() to support the generic ml ppl command.
 
### Issues Resolved
https://github.com/opensearch-project/ml-commons/issues/448
 
### Check List
- [ ] New functionality includes testing.
  - [ ] All tests pass
- [ ] New functionality has been documented.
  - [ ] New functionality has javadoc added
- [ ] Commits are signed per the DCO using --signoff 

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.
For more information on following Developer Certificate of Origin and signing off your commits, please check [here](https://github.com/opensearch-project/OpenSearch/blob/main/CONTRIBUTING.md#developer-certificate-of-origin).

### CODE REVIEW

File: client/src/main/java/org/opensearch/ml/client/MachineLearningNodeClient.java [hunk 1]
```diff
@@ -169,4 +229,77 @@ private void validateMLInput(MLInput mlInput, boolean requireInput) {
...
+                        BATCH_RCF : FIT_RCF;
+            default:
+                throw new IllegalArgumentException(
+                        String.format("unsupported algorithm: %s.", algo));
+        }
+    }
+
+    private MLAlgoParams convertArgumentToMLParameter(Map<String, Object> arguments,
+                                                        FunctionName func) {
+        switch (func) {
```
Comment (ylwu-amzn) [inline client/src/main/java/org/opensearch/ml/client/MachineLearningNodeClient.java:255]: For adding new algorithms, we need to change this client part as well. How about moving this part inside plugin or ml-algorithm , so we keep client part stable.

File: client/src/main/java/org/opensearch/ml/client/MachineLearningClient.java [hunk 2]
```diff
@@ -87,6 +89,26 @@ default ActionFuture<MLOutput> train(MLInput mlInput, boolean asyncTask) {
      */
     void train(MLInput mlInput, boolean asyncTask, ActionListener<MLOutput> listener);
 
+    /**
+     * Execute train/predict/trainandpredict.
+     * @param mlInput ML input
+     * @param args algorithm parameters
+     * @return ActionFuture of MLOutput
+     */
+    default ActionFuture<MLOutput> execute(MLInput mlInput, Map<String, Object> args) {
```
[the remark this answers is not in the document]
Comment (jngz-es) [reply client/src/main/java/org/opensearch/ml/client/MachineLearningClient.java]: Renamed to run.

File: client/src/main/java/org/opensearch/ml/client/MachineLearningNodeClient.java (hunk 1 above)
Comment (jngz-es) [reply client/src/main/java/org/opensearch/ml/client/MachineLearningNodeClient.java]: Good suggestion, moved to common part as client only depends on common part.

File: common/src/main/java/org/opensearch/ml/common/input/Constants.java [hunk 3]
```diff
@@ -0,0 +1,33 @@
+/*
+ * Copyright OpenSearch Contributors
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+package org.opensearch.ml.common.input;
+
+public class Constants {
```
Comment (ylwu-amzn) [inline common/src/main/java/org/opensearch/ml/common/input/Constants.java:8]: Seems this class is duplicate of MLConstants. Move to `common` package?

Comment (jngz-es) [reply common/src/main/java/org/opensearch/ml/common/input/Constants.java:8]: It is not duplicate. I separated the constants to two parts. The constant like "train" should be only used by the client. Anyway, I can put them together into the commom.

**Review by ylwu-amzn — APPROVED** [review]

**Pull request merged**